src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -65,33 +74,72 @@ protected override bool ValidateParameters()
         {
             _createPDBCommand = CompilationEntry.GetMetadata(MetadataKeys.CreatePDBCommand);
 
-            if (CrossgenTool == null && Crossgen2Tool == null)
-            {
-                return false;
-            }
             if (IsPdbCompilation && CrossgenTool == null)
             {
-                // We need the crossgen tool for now to emit native symbols. Crossgen2 does not yet support this feature
+                // PDB compilation is a step specific to Crossgen1 and 5.0 Crossgen2
+                // which didn't support PDB generation. 6.0  Crossgen2 produces symbols
+                // directly during native compilation.
+                Log.LogError(Strings.CrossgenToolMissingInPDBCompilationMode);
                 return false;
             }
 
-            if (CrossgenTool != null)
+            if (UseCrossgen2)
             {
-                if (!File.Exists(CrossgenTool.ItemSpec) || !File.Exists(CrossgenTool.GetMetadata(MetadataKeys.JitPath)))
+                // We expect JitPath to be set for .NET 5 and {TargetOS, TargetArch} to be set for .NET 6 and later
+                string jitPath = Crossgen2Tool.GetMetadata(MetadataKeys.JitPath);
+                if (Crossgen2Tool == null || string.IsNullOrEmpty(Crossgen2Tool.ItemSpec))
```

Comment:
Nit: In general we do not check `ItemSpec`s for `IsNullOrEmpty`, because it would require some special efforts to create and pass such an `ITaskItem`.  For instance, in this function we do not check that this field is not `IsNullOrEmpty`:
```C#
_outputR2RImage = CompilationEntry.ItemSpec
```

Reply:
Removed in 6th commit.